### PR TITLE
[Chore] separate the runtime, recce state and db models

### PR DESCRIPTION
**Pull request by popcornylu**
Per the decision of the [figma](https://www.figma.com/file/cVGktKTP2OgbpfDgsiKYzl/%5BArchitecture%5D-Support-recce-run-%26-recce-server?type=whiteboard&node-id=0-1&t=6WVG2w28PW96MjgF-0), 

1. The `RecceState` is just a data object for serialize and deserialize
2. The `models/*`  is to put the in-memory data object(DO) and data access object(DAO)
3.  The logic or state export/import moved to the DbtContext object
4. Put the git/pull_request information in the state file
5. Add `/api/info` endpoint to replace `/api/lineage` and it can get all necessary information by one API call.

**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Chore

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

**Comment by kentwelcome** [comment]
The legacy state file version generated by `recce run` v0.11.0 should be handled. At least, a user-readable error message should be thrown. For example, rerun the `recce run` with the latest version of recce.

File: recce/server.py [hunk 1]
```diff
@@ -109,9 +104,18 @@ async def health_check(request: Request):
...
+            'lineage': {
+                'base': dbt_context.get_lineage(base=True),
+                'current': dbt_context.get_lineage(base=False),
+            },
+        }
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=str(e))
 
 
 @app.get("/api/lineage")
```
Comment (kentwelcome) [inline recce/server.py]: Seem we replace the `/api/lineage` by `/api/info`.
Can just remove the `GET /api/lineage` from our code base.
Or mark it as deprecated API and plan to remove the related code in the future.

**Review by kentwelcome — APPROVED** [review]
LGTM
Only 2 minor issues:
 - Throw message when processing the legacy state file
 - Remove or mark deprecated the related code of the GET API `/api/lineage`

**Comment by popcornylu** [comment]
> Throw message when processing the legacy state file
```

I makes it failed fast and throw error message.
$ recce server --review /tmp/recce_state_11.json
[Error] Failed to launch server due to:
No enough dbt artifacts in the state file. Please use the latest recce to generate the recce state
```

> Remove or mark deprecated the related code of the GET API /api/lineage
Removed

**Pull request merged**